engine/src/main/java/org/terasology/utilities/collection/CircularToggleSet.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.utilities.collection;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * CircularToggleSet is a "toggleable" doubly-linked list, provides uniqueness with the help of HashMap.
+ * <p>
+ * CircularToggleSet's {@link #toggle()} provides iteration through out given elements. Whenever, {@link #toggle()} is called
+ * first-inserted element is returned. Therefore, iteration starts from beginning again.
+ *
+ * @param <T>
+ */
+public class CircularToggleSet<T> {
```

Comment:
I don't understand why you added this class. What was wrong with the LinkedHashSet? You needed to regenerate its iterator when modified, but as @msteiger mentioned that's not an expensive operation and the modifications do not occur in hot code anyway. Am I missing something?